src/lib/assetBridger/l1l3Bridger.ts

```diff
@@ -0,0 +1,1141 @@
+}
+
+/**
+ * Base functionality for L1 to L3 bridging.
+ */
+class BaseL1L3Bridger {
+  public readonly l1Network: L1Network
+  public readonly l2Network: L2Network
+  public readonly l3Network: L2Network
+
+  public readonly defaultGasPricePercentIncrease: BigNumber =
+    BigNumber.from(200)
+
+  constructor(l3Network: L2Network) {
+    const l2Network = l2Networks[l3Network.partnerChainID]
+    if (!l2Network) {
+      throw new ArbSdkError(
+        `Unknown l2 network chain id: ${l3Network.partnerChainID}`
+      )
+    }
+
+    const l1Network = l1Networks[l2Network.partnerChainID]
+    if (!l1Network) {
+      throw new ArbSdkError(
+        `Unknown l1 network chain id: ${l2Network.partnerChainID}`
+      )
+    }
+
+    this.l1Network = l1Network
+    this.l2Network = l2Network
+    this.l3Network = l3Network
+  }
+
+  /**
+   * Check the signer/provider matches the l1Network, throws if not
+   * @param sop
+   */
+  protected async _checkL1Network(sop: SignerOrProvider): Promise<void> {
+    await SignerProviderUtils.checkNetworkMatches(sop, this.l1Network.chainID)
+  }
+
+  /**
+   * Check the signer/provider matches the l2Network, throws if not
+   * @param sop
+   */
+  protected async _checkL2Network(sop: SignerOrProvider): Promise<void> {
+    await SignerProviderUtils.checkNetworkMatches(sop, this.l2Network.chainID)
+  }
+
+  /**
+   * Check the signer/provider matches the l3Network, throws if not
+   * @param sop
+   */
+  protected async _checkL3Network(sop: SignerOrProvider): Promise<void> {
+    await SignerProviderUtils.checkNetworkMatches(sop, this.l3Network.chainID)
+  }
+
+  protected _percentIncrease(num: BigNumber, increase: BigNumber): BigNumber {
+    return num.add(num.mul(increase).div(100))
+  }
+}
+
+/**
+ * Base functionality for bridging ERC20 tokens from L1 to L3.
+ */
+class BaseErc20L1L3Bridger extends BaseL1L3Bridger {
+  public readonly teleporterAddresses: TeleporterAddresses
+
+  // todo: tune these
+  public readonly defaultRetryableGasParams: ManualRetryableGasParams = {
+    l2ForwarderFactoryGasLimit: BigNumber.from(1_000_000),
+    l1l2TokenBridgeGasLimit: BigNumber.from(1_000_000),
+    l2l3TokenBridgeGasLimit: BigNumber.from(1_000_000),
+    l1l2TokenBridgeRetryableSize: BigNumber.from(1000),
+    l2l3TokenBridgeRetryableSize: BigNumber.from(1000),
+  } as const
+
+  public constructor(public readonly l3Network: L2Network) {
+    super(l3Network)
+
+    if (!this.l2Network.teleporterAddresses) {
+      throw new ArbSdkError(
+        `L2 network ${this.l2Network.name} does not have teleporter contracts`
+      )
+    }
+
+    this.teleporterAddresses = this.l2Network.teleporterAddresses
+  }
+
+  /**
+   * Get the corresponding L2 token address for the provided L1 token
+   */
+  public getL2ERC20Address(
+    erc20L1Address: string,
+    l1Provider: Provider
+  ): Promise<string> {
+    return new Erc20Bridger(this.l2Network).getL2ERC20Address(erc20L1Address, l1Provider)
```

Comment:
instead of instantiate a Erc20Bridger everytime, maybe we keep a singleton instance?